server/log_rpc_server.go

```diff
@@ -406,26 +406,64 @@ func (t *TrillianLogRPCServer) GetConsistencyProof(ctx context.Context, req *tri
 func (t *TrillianLogRPCServer) GetLatestSignedLogRoot(ctx context.Context, req *trillian.GetLatestSignedLogRootRequest) (*trillian.GetLatestSignedLogRootResponse, error) {
 	ctx, span := spanFor(ctx, "GetLatestSignedLogRoot")
 	defer span.End()
-	tree, ctx, err := t.getTreeAndContext(ctx, req.LogId, optsLogRead)
+	tree, hasher, err := t.getTreeAndHasher(ctx, req.LogId, optsLogRead)
 	if err != nil {
 		return nil, err
 	}
+	ctx = trees.NewContext(ctx, tree)
 	tx, err := t.registry.LogStorage.SnapshotForTree(ctx, tree)
 	if err != nil {
 		return nil, err
 	}
 	defer tx.Close()
 
-	signedRoot, err := tx.LatestSignedLogRoot(ctx)
+	slr, err := tx.LatestSignedLogRoot(ctx)
 	if err != nil {
 		return nil, err
 	}
 
-	if err := t.commitAndLog(ctx, req.LogId, tx, "GetLatestSignedLogRoot"); err != nil {
+	var root types.LogRootV1
```

Comment:
Does it make sense to call the other API here or refactor the common bits into a function?